src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp

```diff
@@ -1150,6 +1307,111 @@ ParamBlockDesc2 g_param_block_desc(
         p_accessor, &g_pblock_accessor,
     p_end,
 
+     // --- Parameters specifications for SPPM rollup ---
+
+    ParamIdSPPMPhotonType, L"photon_type", TYPE_INT, P_TRANSIENT, 0,
+        p_ui, ParamMapIdSPPM, TYPE_INT_COMBOBOX, IDC_COMBO_SPPM_PHOTON_TYPE,
+        2, IDS_RENDERERPARAMS_SPPM_PHOTON_TYPE_1, IDS_RENDERERPARAMS_SPPM_PHOTON_TYPE_2,
+        p_default, 1,
+        p_accessor, &g_pblock_accessor,
+    p_end,
+
+    ParamIdSPPMDirectLightingType, L"direct_lighting_type", TYPE_INT, P_TRANSIENT, 0,
+        p_ui, ParamMapIdSPPM, TYPE_INT_COMBOBOX, IDC_COMBO_SPPM_DL_TYPE,
+        3, IDS_RENDERERPARAMS_SPPM_DL_TYPE_1, IDS_RENDERERPARAMS_SPPM_DL_TYPE_2, IDS_RENDERERPARAMS_SPPM_DL_TYPE_3,
+        p_default, 0,
+        p_accessor, &g_pblock_accessor,
+    p_end,
+
+    ParamIdSPPMEnableCaustics, L"enable_caustics", TYPE_BOOL, P_TRANSIENT, 0,
+        p_ui, ParamMapIdSPPM, TYPE_SINGLECHEKBOX, IDC_CHECK_SPPM_CAUSTICS,
+        p_default, TRUE,
+        p_accessor, &g_pblock_accessor,
+    p_end,
+
+    ParamIdSPPMEnableImageBasedLighting, L"enable_image_based_lighting", TYPE_BOOL, P_TRANSIENT, 0,
+        p_ui, ParamMapIdSPPM, TYPE_SINGLECHEKBOX, IDC_CHECK_SPPM_IMAGE_BASED_LIGHTING,
+        p_default, TRUE,
+        p_accessor, &g_pblock_accessor,
+    p_end,
+
+    ParamIdSPPMPhotonTracingMaxBounces, L"photon_tracing_max_bounces", TYPE_INT, P_TRANSIENT, 0,
+        p_ui, ParamMapIdSPPM, TYPE_SPINNER, EDITTYPE_INT, IDC_TEXT_SPPM_PT_BOUNCES, IDC_SPINNER_SPPM_PT_BOUNCES, SPIN_AUTOSCALE,
+        p_default, -1,
+        p_range, -1, 100,
```

Comment:
Why can we set Max Bounces to -1 if there's already a checkbox to enable/disable max bounce limits?

Reply:
Changed it so that when max bounce limit is enabled the default is 8, range 0 to 100. Only when max bounce limits is disabled it will be set to -1 (unlimited).